controllers/extparticipant/utils.js

```diff
@@ -47,11 +47,29 @@ const serialize_attributes = function serialize_attributes(a) {
   return `${oid}=${value}`;
 };
 
-exports.verify_certificate_chain = function verify_certificate_chain(chain) {
-  return forge.pki.verifyCertificateChain(root_ca_store, chain);
+const cert_is_trusted = function cert_is_trusted(cert) {
+  const trusted_list = get_trusted_list();
+  const fingerprint = new crypto.X509Certificate(forge.pki.certificateToPem(cert)).fingerprint256.replaceAll(':', '');
+  return trusted_list.includes(fingerprint);
 };
 
-exports.validate_client_certificate = function validate_client_certificate(chain) {
+exports.verify_certificate_chain = async function verify_certificate_chain(chain) {
+  // Verify the certificate chain one by one
+  let result = true;
+  for (let i = 0; i < chain.length - 2 && result; i++) {
+    const subject = chain[i];
+    if (cert_is_trusted(subject)) {
+      debug(`Certificate ${i} is trusted`);
+      return result;
+    }
+    const issuer = chain[i + 1];
+    result = result && issuer.verify(subject);
+  }
+  result = result && (await cert_is_trusted(chain[chain.length - 1]));
```

Comment:
If cert_is_trusted is no longer an async function, you don't need the await here, so this function does not need to be async, so calling this function does not require await, etc